<launch>

  <arg name="laserscan_generator_param"  default="$(find-pkg-share laserscan_generator)/config/laserscan_generator_node.param.yaml"/>
  <arg name="csv_path" default="$(find-pkg-share laserscan_generator)/map/lane.csv"/>

  <arg name="use_sim_time" default="true"/>
  <arg name="pose_topic" default="/localization/pose_with_covariance"/>
  <arg name="scan_topic" default="/sensing/lidar/scan"/>
  
  <node pkg="tf2_ros"
        exec="static_transform_publisher"
        name="static_tf_pub_laser"
        args="0.0 0.0 0.0 0 0 0 sensor_kit_base_link laser"
        >
    <param name="use_sim_time" value="$(var use_sim_time)"/>
  </node>
  
  <node pkg="laserscan_generator"
        exec="laserscan_generator_node"
        name="laserscan_generator_node"
        output="screen">

    <param from="$(var laserscan_generator_param)"/>
    <param name="csv_path" value="$(var csv_path)"/>
    
    <param name="use_sim_time" value="$(var use_sim_time)"/>

    <remap from="/localization/pose_with_covariance" to="$(var pose_topic)"/>
    <remap from="/scan" to="$(var scan_topic)"/>
    
  </node>

</launch>
